cpp/src/c_api/array.hpp

```diff
@@ -87,15 +86,38 @@ struct cugraph_type_erased_host_array_view_t {
     return reinterpret_cast<T*>(data_);
   }
 
+  template <typename T>
+  T const* as_type() const
+  {
+    return reinterpret_cast<T const*>(data_);
+  }
+
   size_t num_bytes() const { return num_bytes_; }
 };
 
 struct cugraph_type_erased_host_array_t {
-  std::unique_ptr<std::byte[]> data_;
-  size_t size_;
-  size_t num_bytes_;
+  // NOTE: size must be first here because the device buffer is released
```

Comment:
I don't understand this comment, and could you elaborate this comment? It seems like there is more context to this.

Reply:
Comment was copied from above.  In the type_erased_device_buffer we do a release from the `rmm::device_uvector` which changes the size of the vector to 0.  It is important in that case that we copy the size before we call release.

Should delete this comment, std::vector doesn't support release, we do a copy instead.

Reply:
Gotcha, I can see that from the code of `cugraph_type_erased_device_array_t`. No idea why this is necessary (because it is not...)for `cugraph_type_erased_host_array_t` reading the code. Better delete the comment to avoid confusion.